nbsmoke/lint/__init__.py

```diff
@@ -0,0 +1,101 @@
+"""
+Extend pytest to allow flake checking of notebooks.
+
+A notebooks is first converted to a python script, then checked with
+pyflakes.
+
+IPython magics are supposed to be handled in a useful way, being
+converted to python or skipped as appropriate (depending on their
+interaction with python code). See the magics module for more info.
+"""
+
+import sys
+import warnings
+import io
+import os
+import re
+
+import pytest
+import nbformat
+import nbconvert
+
+from . import _pyflakes
+from . import magics
+
+# list of regexes
+flakes_to_ignore = []
+
+class NBLintError(Exception):
+    pass
+
+class LintNb(pytest.Item):
+
+    def repr_failure(self, excinfo):
+        if excinfo.errisinstance(NBLintError):
+            return excinfo.value.args[0]
+        else:
+            return super(LintNb, self).repr_failure(excinfo)
+
+    def runtest(self):
+        with io.open(self.name,encoding='utf8') as nbfile:
+            nb = nbformat.read(nbfile, as_version=4)
+
+            magics_thing = magics.Thing(
+                extra_line_blacklist = _get_list_from_conf('nbsmoke_flakes_line_magics_blacklist', self.parent.parent.config),
+                extra_cell_blacklist = _get_list_from_conf('nbsmoke_flakes_cell_magics_blacklist', self.parent.parent.config))
+            magics_thing.insert_get_ipython(nb)
+
+            ipy, _ = nbconvert.PythonExporter().from_notebook_node(nb)
+
+            debug=self.config.option.nbsmoke_lint_debug
+            filenames = []
+
+            self._write_debug_file(debug, ipy, self.name, "pre", filenames)
+
+            py = magics_thing.ipython_to_python_for_flake_checks(ipy)
+
+            self._write_debug_file(debug, py, self.name, "post", filenames)
+
+            flake_result = _pyflakes.flake_check(
+                # notebooks will start with "coding: utf-8", but py already unicode
+                py.encode('utf8') if sys.version_info[0]==2 else py,
+                self.name)
+
+            ### remove flakes by regex
+            _user_flakes_to_ignore = self.parent.parent.config.getini('nbsmoke_flakes_to_ignore')
+            if _user_flakes_to_ignore != '':
+                _user_flakes_to_ignore = _user_flakes_to_ignore.splitlines()
+            for pattern in set(flakes_to_ignore) | set(_user_flakes_to_ignore):
+                flake_result['messages'] = [msg for msg in flake_result['messages'] if not re.search(pattern, msg)]
+            ###
```

Comment:
Is there meant to be a comment here?

Reply:
It marks the end of a comment block (good old commented chunk instead of a function...). Have clarified.